src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs

```diff
@@ -422,14 +424,14 @@ public async Task MergeDependencyPullRequestAsync(string pullRequestUrl, MergePu
         public async Task CreateOrUpdatePullRequestCommentAsync(string pullRequestUrl, string message)
         {
             (string owner, string repo, int id) = ParsePullRequestUri(pullRequestUrl);
-            IssueComment lastComment = (await Client.Issue.Comment.GetAllForIssue(owner, repo, id)).LastOrDefault();
+            IssueComment lastComment = (await ExecuteAndLogApiErrors(() => Client.Issue.Comment.GetAllForIssue(owner, repo, id))).LastOrDefault();
             if (lastComment != null && lastComment.Body.EndsWith(CommentMarker))
             {
-                await Client.Issue.Comment.Update(owner, repo, lastComment.Id, message + CommentMarker);
+                await ExecuteAndLogApiErrors(() => Client.Issue.Comment.Update(owner, repo, lastComment.Id, message + CommentMarker));
             }
             else
             {
-                await Client.Issue.Comment.Create(owner, repo, id, message + CommentMarker);
+                await ExecuteAndLogApiErrors(() => Client.Issue.Comment.Create(owner, repo, id, message + CommentMarker));
```

Comment:
Why make all these calls catch and log? Why not just log at the higher level?  Catching/logging should always be done at the "highest" level possible, so that you don't have to wrap every single call to every method like this.

Reply:
I had a nice grep :-)

But yeah, for these methods that make more than one call with the GitHub client it's probably nicer to just have one try-catch.

Reply:
Even that's probably not high enough.  I'd put it WAAAAY out at the entry point, and only have a single try/catch anywhere.

Reply:
Fine point. I originally didn't like that a higher-level class would now need to know about a low-level (and implementation-specific) Octokit class, but the distance really isn't much at all and the wins in simplicity are clear. So I moved this up to the top level.

Reply:
Yeah, it's an... interesting problem.  C# doesn't have a good pattern of "I don't want to leak the details of this exception out to my callers" without try/catch everywhere.

If you wanted to get REALLY fancy, you could twist this whole things on it's head, and rather than changing what you log, instead inject something into the Logging/AppInsights pipeline to log this sort of thing special (sort of like we do for the HttpClient in all our AzDO calls).

Basically make it so that every exception of this type always gets logged fancy.  You could use an ITelemetryInitializer to handle this, and it could just get injected into the AI pipeline at the same place other GitHub stuff is.